.github/workflows/images.yml

```
@@ -48,6 +48,90 @@ jobs:
           cache-to: type=gha,mode=max
           file: gobot/Containerfile
 
+  push_to_registries_ui:
+    name: Push UI container image to GHCR
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: docker/setup-buildx-action@v3
+
+      - name: Extract metadata (tags, labels) for UI image
+        id: ui_meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}/bot-ui
+
+      - name: Build and push ui image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            GITHUB_USER=instructlab-bot
+            GITHUB_TOKEN=${{ secrets.BOT_GITHUB_TOKEN }}
+          target: ui
+          push: true
+          tags: ${{ steps.ui_meta.outputs.tags }}
+          labels: ${{ steps.ui_meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          file: ui/Containerfile
+
+  push_to_registries_ui:
+    name: Push apiserver container image to GHCR
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: docker/setup-buildx-action@v3
+
+      - name: Extract metadata (tags, labels) for apiserver image
+        id: apiserver_meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}/apiserver
+
+      - name: Build and push apiserver image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            GITHUB_USER=instructlab-bot
+            GITHUB_TOKEN=${{ secrets.BOT_GITHUB_TOKEN }}
+          target: ui
+          push: true
+          tags: ${{ steps.apiserver_meta.outputs.tags }}
+          labels: ${{ steps.apiserver_meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          file: ui/apiserver/Containerfile
+
```

In our workflows we found this awesome action that will let you SSH to the runner if there is an issue, I suggest we add this in an `on failure` type clause:
```suggestion
      - name: Setup tmate session
        if: ${{ failure() }}
        uses: mxschmitt/action-tmate@v3.18
        timeout-minutes: 15
        with:
          detached: false
          limit-access-to-actor: true
```
This way if something breaks you can interact with stuff directly, and it will only spin-up on failure

---

This looks awesome, i think we should do it across our workflows. Want to pop an issue and hack on it?